src/wp-includes/category-template.php

```diff
@@ -1337,7 +1337,7 @@ function get_the_term_list( $post_id, $taxonomy, $before = '', $sep = '', $after
 		if ( is_wp_error( $link ) ) {
 			return $link;
 		}
-		$links[] = '<a href="' . esc_url( $link ) . '" rel="tag">' . $term->name . '</a>';
+		$links[] = '<a class="' . esc_attr( $taxonomy ) . ' ' . esc_attr( $term->slug ) . '" href="' . esc_url( $link ) . '" rel="tag">' . $term->name . '</a>';
```

Comment:
```suggestion
		$links[] = '<a class="' . esc_attr( sanitize_html_class( $taxonomy ) ) . ' ' . esc_attr( sanitize_html_class( $term->slug ) ) . '" href="' . esc_url( $link ) . '" rel="tag">' . $term->name . '</a>';
```

This would ensure only a valid HTML class is used. We could also move the `$taxonomy` sanitization out of the `foreach` and store it in a variable to prevent sanitizing on every iteration of the loop.